### ISSUE TITLE
Adapting Model for Multiple Outputs

### ISSUE THREAD

**Issue by victoic** [open-mmlab/mmdetection#5493]
Hello,
I'm implementing a modification over the Faster RCNN user on the MMDet_Tutorial.
My question is: **_given I want to output bbox/mask plus multiple classifications per ROI, what would be the reccomended way of implementation?_**
I created a BBoxHead class based on the ConvFCBBoxHead in which the cls_score is returned as a hstack of the classfication predictions for each attribute. However this modification in cls_score shape seems to break, as expected, the mmdet pipeline and I can't find what else would I need to modify.
So, I'm left wondering if there is a tutorial or example of something similar I could use. Do I need to create multiple heads for each classification? But if so, how do I set a conjunct loss function to use all of the classes?

**Comment by AronLin** [comment]
You can set `cfg=None` into simple_test_bboxes() in standard_roi_head or bbox_head.get_bboxes() in bbox_head and you will get `(bboxes, scores)` without NMS.

The `scores` will be with the shape of `(K, num_classes)`. Is this what you need?

**Comment by victoic** [comment]
I don't think so. Think of like the E2EMLT model, where you have bbox prediction and cls_score but also a rotation angle prediction. What I'm trying to accomplish is one ROI having the bbox, and multiple cls_scores. For example: cls_score1 for label_class, csl_score2 for object color class, cls_score3 for object distance from camera, etc.

**Comment by PeterVennerstrom** [comment]
A single head should be sufficient.  To add extra predicted attributes extra labels need to be fed into the model training and assembled as targets.  The head architecture needs to output additional values and this output needs to be split by task then run through activation functions and loss calculated as appropriate.  The new losses can be labeled and returned like the original losses (key/value pairs in a dictionary returned by the head's loss() method) and will be automatically added together and logged in the training loop.  

E.g. Color class may use softmax --> cross entropy loss and distance to camera might use Relu --> L1 loss
